extensions/interactions/RatioExpressionInput/directives/ratio-expression-input-validation.service.ts

```diff
@@ -0,0 +1,133 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Validator service for the RatioExpressionInput interaction.
+ */
+
+import { Injectable } from '@angular/core';
+import { downgradeInjectable } from '@angular/upgrade/static';
+
+import { AnswerGroup } from
+  'domain/exploration/AnswerGroupObjectFactory';
+import { Warning, baseInteractionValidationService } from
+  'interactions/base-interaction-validation.service';
+import { RatioExpressionInputCustomizationArgs } from
+  'extensions/interactions/customization-args-defs';
+import { RatioObjectFactory } from
+  'domain/objects/RatioObjectFactory';
+import { RatioExpressionInputRulesService } from
+  './ratio-expression-input-rules.service';
+import { Outcome } from
+  'domain/exploration/OutcomeObjectFactory';
+import { AppConstants } from 'app.constants';
+
+
+@Injectable({
+  providedIn: 'root'
+})
+export class RatioExpressionInputValidationService {
+  constructor(
+    private rof: RatioObjectFactory,
+    private baseInteractionValidationServiceInstance:
+      baseInteractionValidationService) {}
+
+  getCustomizationArgsWarnings(
+      customizationArgs: RatioExpressionInputCustomizationArgs): Warning[] {
+    return [];
+  }
+
+  getAllWarnings(
```

Comment:
Should there be a validation warning for cases where the number of terms in some of the creator's inputs is lesser than the minimum number of terms required from the learner since that rule input would never match?

Reply:
yes I have added that in a later commit